trieste/models/model_interfaces.py

```diff
@@ -94,6 +94,95 @@ def optimize(self, dataset: Dataset) -> None:
         raise NotImplementedError
 
 
+class ModelStack(TrainableProbabilisticModel):
+    r"""
+    A :class:`ModelStack` is a wrapper around a number of :class:`TrainableProbabilisticModel`\ s.
+    It combines the outputs of each model for predictions and sampling, and delegates training data
+    to each model for updates and optimization.
+
+    **Note:** Only supports vector outputs (i.e. with event shape [E]). Outputs for any two models
+    are assumed independent. Each model may itself be single- or multi-output, and any one
+    multi-output model may have dependence between its outputs.
+    """
+
+    def __init__(
+        self,
+        model_with_event_size: tuple[TrainableProbabilisticModel, int],
+        *models_with_event_sizes: tuple[TrainableProbabilisticModel, int],
+    ):
+        r"""
+        The order of individual models specified at :meth:`__init__` determines the order of the
+        :class:`ModelStack` output dimensions.
+
+        :param model_with_event_size: The first model, and the size of its output events.
+            **Note:** This is a separate parameter to ``models_with_event_sizes`` simply so that the
+            method signature requires at least one model. It is not treated specially.
+        :param \*models_with_event_sizes: The other models, and sizes of their output events.
+        """
```

Comment:
I think I like this idea (if I have understood it correctly)

Reply:
A ModelStack would be defined once when setting up the BO loop. It wouldn't be deconstructed (indeed there's no way to get the models out once they're in) so there's no chance for it to be mixed up. The user would have to make sure they put the models in the same order as they've put their observations in the observer